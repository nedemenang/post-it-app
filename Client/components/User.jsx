import React, { Component } from 'react';
import '../public/style.css';
import { addUserToGroup } from '../actions/AppActions';
import { ListItem } from 'material-ui/List';
import Avatar from 'material-ui/Avatar';


const style = { margin: 5 };
/**
 * @class User
 * @extends {Component}
 */
class User extends Component {

  /**
   * Calls the add user to group action
   * @return {void} return void
   * @memberof User
   */
  userClicked() {
    if (confirm('Are you sure you want to add this user to group?') === true) {
      if (this.props.selectedGroup.groupId !== undefined) {
        const userObject = {
          email: this.props.user.email,
          userId: this.props.user.id,
          username: this.props.user.username,
          groupId: this.props.selectedGroup.groupId,
          groupName: this.props.selectedGroup.groupname
        };
        addUserToGroup(userObject);
      }
    }
  }

  /**
   * Creates an instance of User.
   * @param {object} props props object
   * @memberof User
   */
  constructor(props) {
    super(props);
    this.state = {};
    this.userClicked = this.userClicked.bind(this);
  }

  /**
   *
   * Renders the user page
   * @returns {JSX} returns user page
   * @memberof User
   */
  render() {
    return (
        <ListItem leftAvatar={
        <Avatar
          src={this.props.user.profilePic}
          size={30}
          style={style}
           />
      } onTouchTap={this.userClicked}>
         <strong>{this.props.user.username}</strong>
     </ListItem>
    );
  }
}

export default User;
