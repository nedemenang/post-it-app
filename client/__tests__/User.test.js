import User from '../components/User.jsx';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('the user item component', () => {
  let mountedComponent;
  const group = {
    groupId: '-KrAwfM16qbLOig_mSOc',
    groupname: 'Test',
    newMessage: false
  };

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: 'postitappnnam.herokuapp.com/static/files/blank-profile-pic.png'
  };

  const user = {
    userName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    profilePic: 'postitappnnam.herokuapp.com/static/files/blank-profile-pic.png'
  };

  const userItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<User selectedGroup = {group}
      user = {user} loggedInUser = {loggedInUser} key = {1}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = userItem();
    expect(component).toBeDefined();
  });


  it('should render without throwing an error', () => {
    const component = userItem();
    expect(component.contains(user.userName)).toBe(true);
  });

  it('should match snapshot test', () => {
    const component = userItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });


  it('should recieve props', () => {
    expect(Object.keys(userItem().props()).length).toBeGreaterThan(0);
    expect(userItem().instance().props.loggedInUser).toEqual(loggedInUser);
    expect(userItem().instance().props.selectedGroup).toEqual(group);
    expect(userItem().instance().props.user).toEqual(user);
  });
});
