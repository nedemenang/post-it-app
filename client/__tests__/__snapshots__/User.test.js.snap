// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the user item component should match snapshot test 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="No"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
      />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          keyboardFocused={true}
          label="Yes"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Confirmation"
  >
    Are you sure you want to add this user?
  </Dialog>
  <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    leftAvatar={
      <Avatar
        size={30}
        src="postitappnnam.herokuapp.com/static/files/blank-profile-pic.png"
        style={
          Object {
            "margin": 5,
          }
        }
      />
    }
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    open={null}
    primaryTogglesNestedList={false}
    secondaryTextLines={1}
  >
    <strong>
      nnamisco
    </strong>
  </ListItem>
</div>
`;
